src/openqaoa-core/openqaoa/backends/wrapper.py

```diff
@@ -49,6 +63,160 @@ def exact_solution(self, *args, **kwargs):
 
 
+#------------------------ZNE WRAPPER---------------------------------
+available_factories = [
+    "Richardson",
+    "Linear",
+    "Poly",
+    "Exp",
+    "PolyExp",
+    "AdaExp",
+    "FakeNodes"
+]
+available_scaling = [
+    "fold_gates_at_random",
+    "fold_gates_from_right",
+    "fold_gates_from_left"
+]
+class ZNEWrapper(BaseWrapper):
+    """
+    This class inherits from the BaseWrapper and need to be backend agnostic
+    to QAOA implementations on different devices and their respectives SDKs. It 
+    implements Zero Noise Extrapolation (ZNE) from Mitiq framework. ZNE is an 
+    error mitigation technique used to extrapolate the noiseless expectation
+    value of an observable from a range of expectation values computed at 
+    different noise levels.
+
+    Parameters
+    ----------
+    factory: str
+        The name of the object that determines the zero-noise extrapolation method.
+    scaling: str
+        The name of the function for scaling the noise of a quantum circuit.
+    scale_factors: List[str]
+        Sequence of noise scale factors at which expectation values should be measured.
+        For AdaExpFactory, just the first element of the list will be considered.
+    order: int
+        Extrapolation order (degree of the polynomial fit). It cannot exceed len(scale_factors) - 1.
+        Just used for PolyFactory and PolyExpFactory.
+    steps: int
+        The number of optimization steps. At least 3 are necessary.
+        Just used for AdaExpFactory.
+    """
+
+    def __init__(self, backend, factory, scaling, scale_factors, order, steps):
+        super().__init__(backend)
+
+        # only qiskit backends are supported
+        assert( type(backend) == DEVICE_NAME_TO_OBJECT_MAPPER['qiskit.qasm_simulator'] or 
+                type(backend) == DEVICE_NAME_TO_OBJECT_MAPPER['qiskit.shot_simulator'] or
+                type(backend) == DEVICE_NAME_TO_OBJECT_MAPPER['qiskit.statevector_simulator']
+                or type(backend) == QAOAQiskitQPUBackend), "Only Qiskit backends are supported."
+        # for Mitiq integration, is necessary to transpile the circit. Mitiq doesn't support the RZZ gate.
+        self.parametric_circuit = transpile(self.backend.parametric_circuit, basis_gates=["h","rx","cx"])
+
+        assert(factory in available_factories), "Supported factories are: Poly, Richardson, Exp, FakeNodes, Linear, PolyExp, AdaExp"
+        assert(scaling in available_scaling), "Supported scaling methods are: fold_gates_at_random, fold_gates_from_right, fold_gates_from_left"
+        assert(
+            isinstance(scale_factors, list) and 
+            all(isinstance(x, int) and x >= 1 for x in scale_factors)
+            ), "Scale factor must be a list of ints greater or equal to 1"
+            
+        assert(type(order) == int and order >=1),"Order must a int >=1"
+        assert(type(steps) == int),"Order must a int"
+
+        self.factory_obj = None
+        if factory == "Richardson":
```

Comment:
We are doing case testing here, we can raise the error message here instead if the `factory`, `scaling` or any other parameter is not valid

Reply:
We already raise a ValueError previously, if parameters are not valid. I think that raising a further exception here would be redundant. What do you think?

Reply:
What I mean is that the check is redundant in the sense that you first check if your factory is in the available factories line:118 and then you match the string to the supported factories. What you could do is raise a `ValueError` in the last `else` statement.
The same comment can probably be applied to `scale_noise`.